### DISCUSSION TITLE
add_stream(, rate=float type) causes an error in 13.1.0

### DISCUSSION THREAD

**Discussion by nagadomi** [PyAV-Org/PyAV#1572] [4. Bugs] [answered]
```python
import av
from fractions import Fraction

container = av.open("output.mp4", 'w')
stream = container.add_stream("libx264", rate=15.0)
```
```
Traceback (most recent call last):
  File "/data/nunif-tmp/py_av_bug.py", line 5, in <module>
    stream = container.add_stream("libx264", rate=15.0)
  File "av/container/output.pyx", line 104, in av.container.output.OutputContainer.add_stream
  File "av/utils.pyx", line 51, in av.utils.to_avrational
AttributeError: 'float' object has no attribute 'numerator'
```

Works fine in 13.0.0 and older.
 Is this a new spec or a bug?

**Comment by WyattBlue** [comment, marked as answer]
This is a consequence of
>Permit only setting time_base with a Fraction, as mypy is unable to respect different types in getters vs setters.

because they both use the same utility function. While I think it's okay for PyAV to raise this error (framerate is AVRational, not float64), documentation should have been changed and mypy should be flagging this.

**Reply by nagadomi** [reply]
Thanks for the answer.
float values like 0.123 have caused errors before, so I don't think it is that major compatibility issue.

**Reply by lucky9-cyou** [reply]
Hello, I would like to ask how to fix the above problem：

AttributeError: 'float' object has no attribute 'numerator'

**Reply by nagadomi** [reply]
```python
import av
from fractions import Fraction

container = av.open("output.mp4", 'w')
stream = container.add_stream("libx264", rate=Fraction(15.0))  # use Fraction(float_fps_value)
```
